pkg/virt-controller/watch/pod.go

```diff
@@ -41,132 +41,146 @@ func NewPodControllerWithListWatch(vmCache cache.Store, _ record.EventRecorder,
 	return kubecli.NewController(lw, queue, &v1.Pod{}, NewPodControllerFunc(vmCache, restClient, vmService, clientset))
 }
 
-func NewPodControllerFunc(vmCache cache.Store, restClient *rest.RESTClient, vmService services.VMService, clientset *kubernetes.Clientset) kubecli.ControllerFunc {
+func NewPodControllerFunc(vmCache cache.Store, restClient *rest.RESTClient, vmService services.VMService, clientset *kubernetes.Clientset) kubecli.ControllerDispatch {
+	dispatch := PodDispatch{
+		vmCache:    vmCache,
+		restClient: restClient,
+		vmService:  vmService,
+		clientset:  clientset,
+	}
+	return &dispatch
+}
 
-	return func(store cache.Store, queue workqueue.RateLimitingInterface, key interface{}) {
-		// Fetch the latest Vm state from cache
-		obj, exists, err := store.GetByKey(key.(string))
+type PodDispatch struct {
```

Comment:
Consider making this struct private--all the struct members are private so this isn't really usable outside this module anyways.